src/main/features/directions-questionnaire/routes/hearing-location.ts

```diff
@@ -1,9 +1,109 @@
+import { AlternativeCourtOption, HearingLocation } from 'directions-questionnaire/forms/models/hearingLocation'
+import { Draft } from '@hmcts/draft-store-client'
+import { DirectionsQuestionnaireDraft } from 'directions-questionnaire/draft/directionsQuestionnaireDraft'
+import { Court } from 'court-finder-client/court'
+import { CourtFinderClient } from 'court-finder-client/courtFinderClient'
+import { Claim } from 'claims/models/claim'
+import { CourtFinderResponse } from 'court-finder-client/courtFinderResponse'
+import { FormValidator } from 'forms/validation/formValidator'
+import { DraftService } from 'services/draftService'
+import { ErrorHandling } from 'shared/errorHandling'
+import { YesNoOption } from 'models/yesNoOption'
+import { ResponseDraft } from 'response/draft/responseDraft'
+import { MadeBy } from 'offer/form/models/madeBy'
+import { getUsersRole } from 'directions-questionnaire/helpers/directionsQuestionnaireHelper'
+import { User } from 'idam/user'
+import { PartyDetails } from 'forms/models/partyDetails'
+
+function renderPage (res: express.Response, form: Form<HearingLocation>, fallbackPage: boolean) {
+  res.render(Paths.hearingLocationPage.associatedView, { form: form, fallbackPage: fallbackPage, party: getUsersRole(res.locals.claim, res.locals.user) })
+}
+
+async function getNearestCourt (postcode: string): Promise<Court> {
+  const courtFinderClient: CourtFinderClient = new CourtFinderClient()
+  const response: CourtFinderResponse = await courtFinderClient.findMoneyClaimCourtsByPostcode(postcode)
+
+  if (response.statusCode !== 200 || response.courts.length === 0) {
+    return undefined
+  } else {
+    return response.courts[0]
+  }
+}
+
+function getDefaultPostcode (res: express.Response): string {
+  const claim: Claim = res.locals.claim
+  const user: User = res.locals.user
+  if (getUsersRole(claim, user) === MadeBy.DEFENDANT) {
+    const responseDraft: Draft<ResponseDraft> = res.locals.responseDraft
+    const partyDetails: PartyDetails = responseDraft.document.defendantDetails.partyDetails
+    if (partyDetails && partyDetails.address.postcode) {
+      return responseDraft.document.defendantDetails.partyDetails.address.postcode
+    } else {
+      return claim.claimData.defendant.address.postcode
+    }
+  } else {
+    return claim.claimData.claimant.address.postcode
+  }
+}
 
 /* tslint:disable:no-default-export */
 export default express.Router()
-  .get(Paths.hearingLocationPage.uri, (req: express.Request, res: express.Response) => {
-    res.redirect(DashboardPaths.dashboardPage.uri)
+  .get(Paths.hearingLocationPage.uri, async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
+    try {
+      const draft: Draft<DirectionsQuestionnaireDraft> = res.locals.draft
+
+      if (!draft.document.hearingLocation) {
+        const postcode: string = getDefaultPostcode(res)
+        const court: Court = await getNearestCourt(postcode)
+        if (court) {
```

Comment:
The court finder will return nothing if the response code isn't 200, this could be caused mainly by:
-Postcode isn't a for a real location and can't be mapped
-Postcode isn't in a valid format (although this shouldn't happen with our validation)
-The MapIt API court finder depends on is down